src/app/[locale]/onboarding/loading/_hooks/use-logic.ts

```diff
@@ -0,0 +1,180 @@
+"use client";
+
+import { useQuery } from "@tanstack/react-query";
+import i18next from "i18next";
+import { useRouter, useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+import { DEFAULT_PACKAGE_API_URL } from "~/constants";
+import usePackageAPI from "~/hooks/use-package-api";
+import useSQLInit from "~/hooks/use-sql-init";
+import { useAppStore } from "~/stores";
+
+type API = ReturnType<typeof usePackageAPI>;
+
+function useURLParams() {
+  const searchParams = useSearchParams();
+
+  const packageLink = searchParams?.get("packageLink") || undefined;
+  const backendURL = searchParams?.get("backendURL") || undefined;
```

Comment:
```suggestion
  const backendURL = searchParams?.get("backendURL") ?? undefined;
```

Use of `??` would be more appropriate in this situation